### ISSUE TITLE
NoMethodError - undefined method `start_with?' for <Pod::Version version=x.x.x>:Pod::Version

### ISSUE THREAD

**Issue by mzF414** [CocoaPods/cocoapods-downloader#126]
### Command

```
/Users/bundlefile/ruby/2.6.0/bin/pod install --repo-update
```

### Report

* What did you do?

Yesterday everything was perfect, but today, I got this error. I searched the log and found out  the cocoapods-downloader version was changed form 1.5.1 to 1.6.0 .

* What did you expect to happen?

No error 🙏🏻

* What happened instead?

### Stack

```
   CocoaPods : 1.10.1
        Ruby : ruby 2.6.8p205 (2021-07-07 revision 67951) [universal.x86_64-darwin21]
    RubyGems : 3.0.3.1
        Host : macOS 12.2.1 (21D62)
       Xcode : 13.2.1 (13C100)
         Git : git version 2.32.0 (Apple Git-132)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : team-specs - git - https://gitlab.xxx.fm/team/Specs.git @ 19cf70bddeb44766

```

### Plugins

```
cocoapods-binary-flutter              : 0.0.3
cocoapods-deintegrate                 : 1.0.5
cocoapods-plugins                     : 1.0.0
cocoapods-search                      : 1.0.1
cocoapods-trunk                       : 1.6.0
cocoapods-try                         : 1.2.0
```

### Podfile

```ruby
source 'https://gitlab.xxx.fm/team/Specs.git'
# I have ‘2.24.4’ tag on my gitlab(https://gitlab.xxx.fm/team/Specs.git)
AWSCore, ‘2.24.4’
```

### Error

```
NoMethodError - undefined method `start_with?' for <Pod::Version version=2.24.4>:Pod::Version
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-downloader-1.6.0/lib/cocoapods-downloader/git.rb:63:in `block in validate_input'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-downloader-1.6.0/lib/cocoapods-downloader/git.rb:63:in `any?'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-downloader-1.6.0/lib/cocoapods-downloader/git.rb:63:in `validate_input'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-downloader-1.6.0/lib/cocoapods-downloader/git.rb:24:in `preprocess_options'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-downloader-1.6.0/lib/cocoapods-downloader.rb:78:in `preprocess_options'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/downloader.rb:130:in `preprocess_request'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/downloader.rb:37:in `download'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/pod_source_installer.rb:118:in `download_source'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/pod_source_installer.rb:69:in `install!'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:561:in `install_source_of_pod'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:512:in `block (2 levels) in install_pod_sources'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/user_interface.rb:86:in `titled_section'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:511:in `block in install_pod_sources'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:494:in `each'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:494:in `install_pod_sources'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:253:in `block in download_dependencies'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/user_interface.rb:64:in `section'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:252:in `download_dependencies'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:161:in `install!'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/command/install.rb:52:in `run'
/Users/bundlefile/ruby/2.6.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/command.rb:52:in `run'
/Users/bundlefile/ruby/2.6.0/gems/cocoapods-1.10.1/bin/pod:55:in `<top (required)>'
/Users/bundlefile/ruby/2.6.0/bin/pod:23:in `load'
/Users/bundlefile/ruby/2.6.0/bin/pod:23:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/cli/exec.rb:74:in `load'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/cli/exec.rb:74:in `kernel_load'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/cli/exec.rb:28:in `run'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/cli.rb:463:in `exec'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/cli.rb:27:in `dispatch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/cli.rb:18:in `start'
/Library/Ruby/Gems/2.6.0/gems/bundler-1.17.2/exe/bundle:30:in `block in <top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/bundler/friendly_errors.rb:124:in `with_friendly_errors'
/Library/Ruby/Gems/2.6.0/gems/bundler-1.17.2/exe/bundle:22:in `<top (required)>'
/usr/bin/bundle:23:in `load'
/usr/bin/bundle:23:in `<main>'
```

**Comment by orta** [comment]
Should be fixed by #125

**Comment by orta** [comment]
wait, what sort of input is that - did it used to work?

**Comment by tejassharma96** [comment]
having the same error - the previous version worked fine for my project. when can we expect a release with the fix? or should I just pin cocoapods-downloader to 1.5.1 in my Gemfile?

**Comment by dnkoutso** [comment]
Shipped 1.6.1 of cocoapods-downloader can you try again?

**Comment by mzF414** [comment]
> Shipped 1.6.1 of cocoapods-downloader can you try again?

It works! Thank you
